### PR TITLE
Add [fence-effects=""] to stop effect propagation

### PR DESCRIPTION
**Pull request by syg**
This allows for manual fencing where needed, like in GeneratorStart.

### CODE REVIEW

**Comment by bakkot** [comment]
These are intended to be used only at the beginning of steps, yeah? We already have a different mechanism for metadata about steps, namely `id`s (which look like (`1. [id="foo"] Step.`); I'd prefer to reuse that (as in `1. [fences-effects="user-code"] Step.`).

This would require a change to ecmarkdown; I can make that change if this approach sounds good to you.

**Comment by syg** [comment]
> These are intended to be used only at the beginning of steps, yeah? We already have a different mechanism for metadata about steps, namely ids (which look like (1. [id="foo"] Step.); I'd prefer to reuse that (as in 1. [fences-effects="user-code"] Step.).

Exactly right.

> This would require a change to ecmarkdown; I can make that change if this approach sounds good to you.

I admit the only reason I did it this way is because I didn't want to touch ecmarkdown. If you're offering, doing it in ecmarkdown would be ideal.

**Comment by bakkot** [comment]
Rebased on https://github.com/tc39/ecmarkup/pull/392 and switched to the syntax suggested above.

File: src/Xref.ts [hunk 1]
```diff
@@ -91,21 +91,34 @@ export default class Xref extends Builder {
...
+        ) {
+          return false;
+        }
+
+        if (
+          parent
+            .getAttribute('fence-effects')
+            ?.split(',')
+            .map(s => s.trim())
+            .includes(effectName)
```
Comment (syg) [inline src/Xref.ts:119]: I feel like we don't need structured processing here, and a `parent.getAttribute('fence-effects')?.includes()` suffices. WDYT?

Comment (bakkot) [reply src/Xref.ts:119]: Ehh.. that probably does suffice in practice, but it's not like the split + trim is expensive.

Comment (syg) [reply src/Xref.ts:119]: Fair enough, also not that many people build ecma262 anyway.

**Review by bakkot — APPROVED** [review]

**Pull request merged**